server/src/main/java/org/elasticsearch/ExceptionsHelper.java

```diff
@@ -91,7 +92,18 @@ public static Throwable unwrapCause(Throwable t) {
             }
             result = result.getCause();
         }
-        return result;
+
+        // Traverse through the stacktrace and check if the error is caused by NodeNotConnectedException.
```

Comment:
We shouldn't be unwrapping things other than `ElasticsearchWrapperException` here. Do we ever see this exception wrapped inside something that isn't an `ElasticsearchWrapperException`? If so, we should fix that wrapper.

Reply:
If there's an exception that eventually leads to `NodeNotConnectedException` (or a variant of `ConnectTransportException` and I believe there could be many) and it doesn't implement `status()`, won't it end up returning `500` before the propagation even reaches to `NodeNotConnectedException`?

Reply:
Yes, so let's fix that. But not here. This concern applies to all possible root exceptions, not just `NodeNotConnectedException` or `ConnectTransportException`, so it's not appropriate to have some special-case handling for those exceptions here.

- Root exception has wrong `status()` --> fix that exception implementation.
- Wrapper fails to propagate `status()` of inner exception correctly --> fix the wrapper.

> I believe there could be many

I don't think that's the case.